pybossa/redis_lock.py

```diff
@@ -21,12 +21,14 @@
 
 from contributions_guard import ContributionsGuard
 from pybossa.core import sentinel
+from werkzeug.exceptions import BadRequest
 
 TASK_USERS_KEY_PREFIX = 'pybossa:project:task_requested:timestamps:{0}'
 USER_TASKS_KEY_PREFIX = 'pybossa:user:task_acquired:timestamps:{0}'
 TASK_ID_PROJECT_ID_KEY_PREFIX = 'pybossa:task_id:project_id:{0}'
 ACTIVE_USER_KEY = 'pybossa:active_users_in_project:{}'
 EXPIRE_LOCK_DELAY = 5
+EXPIRE_RESERVE_TASK_LOCK_DELAY = 30*60
```

Comment:
Why do we need such a long delay to release reservation? I guess we just need few seconds more than the project timeout time, similar to `EXPIRE_LOCK_DELAY`. 

https://github.com/bloomberg/pybossa/pull/639/files#diff-0696667d581362fe4b56f50b66a6d11691c1c0af039916430c95eb36c1aca302R244 and https://github.com/bloomberg/pybossa/pull/639/files#diff-0696667d581362fe4b56f50b66a6d11691c1c0af039916430c95eb36c1aca302R250

Reply:
Task obtains soft lock on task category upon submitting answer. This is of longer duration than usual `EXPIRE_LOCK_DELAY` so that if user takes short break and return, they still get tasks from reserved category first. We also plan to make this value configurable in future so that project owners can set this value under project settings, and 30 mins would be the default expiration for task reservation.

[This](https://github.com/bloomberg/pybossa/pull/639/files#diff-0696667d581362fe4b56f50b66a6d11691c1c0af039916430c95eb36c1aca302R244) will be applicable if task timed out(not submission).